lxd/db/cluster/update.go

```diff
@@ -70,6 +70,241 @@ var updates = map[int]schema.Update{
+	}
+
+	volumes := make([]struct {
+		ID   int
+		Name string
+	}, count)
+	dest := func(i int) []interface{} {
+		return []interface{}{
+			&volumes[i].ID,
+			&volumes[i].Name,
+		}
+	}
+
+	stmt, err := tx.Prepare(stmts)
+	if err != nil {
+		return errors.Wrap(err, "Failed to prepary storage volume query")
+	}
+
+	err = query.SelectObjects(stmt, dest)
+	if err != nil {
+		return errors.Wrap(err, "Failed to fetch storage volumes")
+	}
+
+	// Remove multiple entries of the same volume when using remote storage
+	for i := 1; i < count; i++ {
+		if volumes[i-1].Name == volumes[i].Name {
+			_, err = tx.Exec(`DELETE FROM storage_volumes WHERE id=?`, volumes[i-1].ID)
+			if err != nil {
+				return errors.Wrap(err, "Failed to delete row from storage_volumes")
+			}
+		}
+	}
+
+	stmts = `
+CREATE TABLE storage_volumes_new (
+    id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,
+    name TEXT NOT NULL,
+    storage_pool_id INTEGER NOT NULL,
+    node_id INTEGER,
+    type INTEGER NOT NULL,
+    description TEXT,
+    project_id INTEGER NOT NULL,
+    content_type INTEGER NOT NULL DEFAULT 0,
+    UNIQUE (storage_pool_id, node_id, project_id, name, type),
+    FOREIGN KEY (storage_pool_id) REFERENCES storage_pools (id) ON DELETE CASCADE,
+    FOREIGN KEY (node_id) REFERENCES nodes (id) ON DELETE CASCADE,
+    FOREIGN KEY (project_id) REFERENCES projects (id) ON DELETE CASCADE
+);`
+
+	// Create new tables where node ID can be null.
+	_, err = tx.Exec(stmts)
+	if err != nil {
+		return err
+	}
+
+	// Copy rows from storage_volumes to storage_volumes_new
+	count, err = query.Count(tx, "storage_volumes", "")
+	if err != nil {
+		return errors.Wrap(err, "Failed to get storage_volumes count")
+	}
+
+	storageVolumes := make([]struct {
+		ID            int
+		Name          string
+		StoragePoolID int
+		NodeID        string
+		Type          int
+		Description   string
+		ProjectID     int
+		ContentType   int
+	}, count)
+
+	dest = func(i int) []interface{} {
+		return []interface{}{
+			&storageVolumes[i].ID,
+			&storageVolumes[i].Name,
+			&storageVolumes[i].StoragePoolID,
+			&storageVolumes[i].NodeID,
+			&storageVolumes[i].Type,
+			&storageVolumes[i].Description,
+			&storageVolumes[i].ProjectID,
+			&storageVolumes[i].ContentType,
+		}
+	}
+
+	stmt, err = tx.Prepare(`SELECT * FROM storage_volumes;`)
+	if err != nil {
+		return errors.Wrap(err, "Failed to prepare storage volumes query")
+	}
+
+	err = query.SelectObjects(stmt, dest)
+	if err != nil {
+		return errors.Wrap(err, "Failed to fetch storage volumes")
+	}
+
+	for _, storageVolume := range storageVolumes {
+		_, err = tx.Exec(`INSERT INTO storage_volumes_new (id, name, storage_pool_id, node_id, type, description, project_id, content_type) VALUES (?, ?, ?, ?, ?, ?, ?, ?);`, storageVolume.ID, storageVolume.Name, storageVolume.StoragePoolID, storageVolume.NodeID, storageVolume.Type, storageVolume.Description, storageVolume.ProjectID, storageVolume.ContentType)
```

Comment:
Please, break this into multiple lines, for readability (especially in the GitHub diff).